snap/hooks/configure

```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+
+DEFAULT_FEATURE_GATES=""
+
+get_feature_gates() {
+  value="$(snapctl get feature-gates)"
+  if [ -z "$value" ]; then
+    value="$DEFAULT_FEATURE_GATES"
+    set_feature_gates "$value"
+  fi
+  echo "$value"
+}
+
+set_feature_gates() {
+  printf "%s" "$1" > "$SNAP_DATA"/feature-gates.conf  
+  snapctl set feature-gates="$1"
+}
+
+
+handle_feature_gates() {
+    feature_gates="$(get_feature_gates)"
+
+    # Validate
+    if ! "$SNAP"/bin/otelcol validate  --feature-gates="$feature_gates" --config="$SNAP/etc/config.yaml" > /dev/null 2>&1; then
+        echo "\"$feature_gates\" is not a valid feature gate definition" >&2
```

Comment:
A non-zero code here does necessarily mean that feature gates are invalid. Could be various reasons, for example:
- snapcraft.yaml bug produced invalid binary
- /etc/config.yaml is missing
- the config is there but invalid unrelated to feature gates

For this reason I think:
1. return non zero for feature gates only if the scope of the check can be narrowed to only feature gates.
2. if we cannot narrow down, then let `config` do `config` and leave it for `snap logs` to show why a restart failed.